testing/src/main/scala/org/http4s/testing/Http4sMatchers.scala

```diff
@@ -19,28 +20,28 @@ trait Http4sMatchers extends Matchers with IOMatchers {
       r.unsafeRunSync.aka("the returned")
     }
 
-  def haveBody[A: EntityDecoder[IO, ?]](a: ValueCheck[A]): Matcher[Message[IO]] =
-    returnValue(a) ^^ { m: Message[IO] =>
+  def haveBody[M[_[_]], A: EntityDecoder[IO, ?]](a: ValueCheck[A])(implicit M: Message[M, IO]): Matcher[M[IO]] =
```

Comment:
Note: these now return the `M[_[_]]` in the return type. Which shows the pattern this allows people to work on safely.